doc/man3/SSL_CTX_set_session_ticket_cb.pod

```diff
@@ -0,0 +1,145 @@
+to the callbacks.
+
+B<gen_cb> is the application defined callback invoked when a session ticket is about
+to be created. The application may call SSL_SESSION_set1_ticket_appdata() at this
+time to add application data to the session ticket. The value of B<arg> is the same
+as that given to  SSL_CTX_set_session_ticket_cb(). The B<gen_cb> callback is defined
+as type B<tls_generate_session_ticket_cb_fn>.
+
+ typedef int (*tls_generate_session_ticket_cb_fn)(SSL *s, void *arg);
+
+B<dec_cb> is the application defined callback invoked after a session ticket has been
+decrypted and any session ticket application data is available. The application may
+call SSL_SESSION_get_ticket_appdata() at this time to retrieve the application data.
+The value of B<arg> is the same as that given to  SSL_CTX_set_session_ticket_cb().
+The B<dec_cb> callback is defined as type B<tls_decrypt_session_ticket_cb_fn>.
+
+ typedef TICKET_RETURN (*tls_decrypt_session_ticket_cb_fn)(SSL *s, SSL_SESSION *ss,
+                                                          const unsigned char *keyname,
+                                                          TICKET_RETURN retv,
+                                                          void *arg);
+
+SSL_SESSION_set1_ticket_appdata() sets the application data specified by B<data> and
+B<len> into B<ss> which is then placed into any generated session tickets. It may be
+called at any time before a session ticket is created to update the data placed into
+the session ticket. However, given that sessions and tickets are created by the
+handshake, the B<gen_cb> is provided to notify the application that a session ticket
+is about to be generated.
+
+SSL_SESSION_get0_ticket_appdata() assigns B<data> to the session ticket application
+data and assigns B<len> to the length of the session ticket application data from
+B<ss>, which may be set via SSL_SESSION_set1_ticket_appdata() or by application data
+in a session ticket. NULL will be assigned to B<data> and 0 will be assigned to B<len>
+if there is no session ticket application data. SSL_SESSION_get0_ticket_appdata() may
+be called any time after a session has been created. The B<dec_cb> is provided to
+notify the application that a session ticket has just been decrypted.
+
+=head1 NOTES
+
+When the B<dec_cb> callback is invoked, the SSL_SESSION B<ss> has not yet been assigned
+to the SSL B<s>. The B<retv> indicates the result of the ticket decription, may be modified
+by the callback before being returned.
+
+When the B<gen_cb> callback is invoked, the SSL_get_session() function may be used to
+retrieve the SSL_SESSION for SSL_SESSION_set1_ticket_appdata().
+
+=head1 RETURN VALUES
+
+The SSL_CTX_set_session_ticket_cb() function does not return a value.
+
+The SSL_SESSION_set1_ticket_appdata() returns 1 on success and 0 on failure.
+
+The B<gen_cb> callback must return 1 to continue the connection. A return of 0 will
+terminate the connection with an INTERNAL_ERROR alert.
```

Comment:
The code this callback is invoked from currently has only one error path; INTERNAL_ERROR is the result. You're right it's not an internal error, but do you have an alternative?